working-with-collections/main.js

```diff
@@ -1,6 +1,105 @@
+/* eslint-disable no-prototype-builtins */
 console.log('Lodash is loaded:', typeof _ !== 'undefined');
 
-// var player1 = {
-//   name: 'name1',
-//   hand: []
-// };
+function createCards() {
+  var cards = [];
+  var ranks = ['Ace', 2, 3, 4, 5, 6, 7, 8, 9, 10, 'Jack', 'Queen', 'King'];
+  var suits = ['diamonds', 'clubs', 'hearts', 'spades'];
+  for (var i = 0; i < 52; i++) {
+    var card = {};
+    card.rank = ranks[i % 13];
+    card.suit = suits[Math.floor(i / 13)];
+    cards.push(card);
+  }
+  return cards;
+}
+
+function createPlayers(numPlayers) {
+  var players = [];
+  for (var i = 0; i < numPlayers; i++) {
+    var player = {};
+    player.name = 'Player ' + (i + 1);
+    player.hand = [];
+    players.push(player);
+  }
+  return players;
+}
+
+function shuffle(totalCards) {
+  while (cardIndexes.length <= totalCards) {
+    var cardIndex = Math.floor(Math.random() * 52);
+    if (!cardIndexes.includes(cardIndex)) {
+      cardIndexes.push(cardIndex);
+    }
+  }
+}
+
+function deal(totalCards, startIndex) {
+  for (var i = startIndex; i < totalCards; i++) {
+    var card = cards[cardIndexes[i]];
+    players[i % players.length].hand.push(card);
+  }
+}
+
+function score() {
+  for (var i = 0; i < players.length; i++) {
+    var score = 0;
+    var name = players[i].name;
+    var hand = players[i].hand;
+    for (var j = 0; j < hand.length; j++) {
+      var card = hand[j];
+      if (card.rank === 'Jack' || card.rank === 'Queen' || card.rank === 'King') {
+        score += 10;
+      } else if (card.rank === 'Ace') {
+        score += 11;
+      } else {
+        score += card.rank;
+      }
+    }
+    if (score > highScore) {
+      winners = [name];
+      highScore = score;
+    } else if (score === highScore) {
+      winners.push(name);
```

Comment:
Again, try to create a _new_ winners array in your `score` function and return it, instead of modifying the global `winners`.